agent/api/task.go

```diff
@@ -156,12 +159,19 @@ type Task struct {
 func (task *Task) PostUnmarshalTask(cfg *config.Config, credentialsManager credentials.Manager) {
 	// TODO, add rudimentary plugin support and call any plugins that want to
 	// hook into this
+	task.initializeContainerDependenciesMap()
 	task.adjustForPlatform(cfg)
 	task.initializeEmptyVolumes()
 	task.initializeCredentialsEndpoint(credentialsManager)
 	task.addNetworkResourceProvisioningDependency(cfg)
 }
 
+func (task *Task) initializeContainerDependenciesMap() {
```

Comment:
Do you really need this method? Why not just do it in `UnmarshalJSON` for `api.Container`?

Reply:
Just realized we are already doing this: 
https://github.com/aws/amazon-ecs-agent/blob/f10cd818e756880d167610ea9489bc44895281c6/agent/api/transitiondependency.go#L50 

Will remove this method.